lib/rubyXL/convenience_methods.rb

```diff
@@ -168,6 +168,10 @@ def modify_fill(style_index, rgb)
     def modify_border(style_index, direction, weight)
       xf = cell_xfs[style_index || 0].dup
       new_border = borders[xf.border_id || 0].dup
+
+      edge = new_border.send(direction)
+      new_border.send("#{direction}=", edge.dup) if edge
```

Comment:
The instance of `BorderEdge` should be duplicated when `new_border` is created otherwise it refers same object and existing border is reused regardless of `new_border.set_edge_style` sets different style.

https://github.com/weshatheleopard/rubyXL/blob/master/lib/rubyXL/objects/border.rb#L41